packages/cli/src/commands/storybook.js

```diff
@@ -77,21 +87,74 @@ export const handler = ({
     require.resolve('@redwoodjs/testing/config/storybook/main.js')
   )
 
-  execa(
-    `yarn ${build ? 'build' : 'start'}-storybook`,
-    [
-      `--config-dir "${storybookConfig}"`,
-      !build && `--port ${port}`,
-      !build && '--no-version-updates',
-      !managerCache && '--no-manager-cache',
-      build && `--output-dir "${buildDirectory}"`,
-      !open && !smokeTest && `--ci`,
-      smokeTest && `--ci --smoke-test`,
-    ].filter(Boolean),
-    {
-      stdio: 'inherit',
-      shell: true,
-      cwd,
+  // Case 1: yarn rw sb
+  // run start-storybook
+  // open in browser (default true)
+  // set port (default 7910)
+  // directory options are hard-coded
+  //NOTE: `--no-manager-cache` option is only available for `start-storybook`
+
+  if (open) {
+    execa(
+      `yarn start-storybook`,
+      [
+        `--config-dir "${storybookConfig}"`,
+        `--port ${port}`,
+        !managerCache && '--no-manager-cache',
+      ].filter(Boolean),
+      {
+        stdio: 'inherit',
+        shell: true,
+        cwd,
+      }
+    )
+  }
+
+  // Case 2: yarn rw sb --build
+  // build-directory option (default web/public/storybook)
+  // directory for config is hard-coded
+  // manager-cache option (default true)
+  //NOTE: `--no-manager-cache` option is only available for `start-storybook`
+
+  if (build) {
+    execa(
+      `yarn build-storybook`,
+      [
+        `--config-dir "${storybookConfig}"`,
+        `--output-dir "${buildDirectory}"`,
+      ].filter(Boolean),
+      {
+        stdio: 'inherit',
+        shell: true,
+        cwd,
+      }
+    )
+  }
+
+  // Case 3: yarn rw sb --smoke-test
+  // runs the start-storybook command; exits gracefully if started successful, throws otherwise
+  // passes --smoke-test and --ci options
+  try {
+    if (smokeTest) {
+      execa(
+        `yarn start-storybook`,
+        [
+          `--config-dir "${storybookConfig}"`,
+          `--port ${port}`,
+          // !managerCache && '--no-manager-cache',
+          `--smoke-test`,
+          `--ci`,
+        ].filter(Boolean),
+        {
+          stdio: 'inherit',
+          shell: true,
+          cwd,
+        }
+      )
     }
-  )
+  } catch (e) {
+    console.log(c.error(e.message))
+    errorTelemetry(process.argv, e.message)
+    process.exit(1)
+  }
```

Comment:
Nice! This is exactly what I had in mind — it's so much cleaner and understandable. I'm ok to sacrifice a bit of DRY for the sake of clarity.

A few things to address:

**1. `yarn start-storybook` is not dependent on the `open` option**
Case 1 is the default, and can accept either --open=true or --open=false.

So the conditionals would be something like this:
```
if (build) {
 // build command and options
} else if (smoke-test) {
 // smoke-test command and options
} else {
 // start-storybook
}

Make sense?

**2. Wrap all cases in `Try { }`**
 Currently, you only have the Try Catch around the 'smoke-test' case. You'll want to extend this to include all the cases.